### ISSUE TITLE
IPIP packet Drop from non-Calico host even I disable IPIP Mode

### ISSUE THREAD

**Issue by hongkunyoo** [projectcalico/calico#4456]
## Expected Behavior
Question & Improvement

I am not an expert in networking. Please understand if my question is absurd.
I need to allow non-Calico traffic on my worker node.

## Current Behavior

It seems dropping IPIP packets from non-Calico hosts rule still works even I disable IPIP Mode.

```bash
$ sudo iptables -L
# ...
Chain cali-INPUT (1 references)
target     prot opt source               destination         
ACCEPT     ipv4 --  anywhere             anywhere             /* cali:PajejrV4aFdkZojI */ /* Allow IPIP packets from Calico hosts */ match-set cali40all-hosts-net src ADDRTYPE match dst-type LOCAL
DROP       ipv4 --  anywhere             anywhere             /* cali:_wjq-Yrma8Ly1Svo */ /* Drop IPIP packets from non-Calico hosts */.   # <--- I want to delete this rule
# ...
```

I manually delete the "Drop IPIP rule" using `iptables` command but it keeps get re-created.

## Possible Solution

Is there a way to remove dropping non-calico IPIP traffic rule?
Or is it fundamentally impossible to change this behavior? 

## Steps to Reproduce (for bugs)

1. Change `ipipMode` to `Never` and replace it

```yaml
cat <<EOF | calicoctl replace -f -
apiVersion: projectcalico.org/v3
kind: IPPool
metadata:
  name: default-ipv4-ippool
spec:
  blockSize: 26
  cidr: 192.168.0.0/16
  ipipMode: Never
  natOutgoing: true
  nodeSelector: all()
EOF
```

```bash
# No tunnel in my route
$ route -n
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.230.208.1    0.0.0.0         UG    10     0        0 eth0
10.0.0.0        10.230.208.1    255.0.0.0       UG    0      0        0 eth0
10.230.208.0    0.0.0.0         255.255.240.0   U     0      0        0 eth0
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
192.168.3.128   10.230.208.1    255.255.255.192 UG    0      0        0 eth0
192.168.47.1    10.230.208.1    255.255.255.255 UGH   0      0        0 eth0
192.168.93.128  10.230.208.1    255.255.255.192 UG    0      0        0 eth0
192.168.148.64  0.0.0.0         255.255.255.192 U     0      0        0 *
192.168.148.65  0.0.0.0         255.255.255.255 UH    0      0        0 cali08044416c96
192.168.154.0   10.230.208.1    255.255.255.192 UG    0      0        0 eth0
192.168.228.64  10.230.208.1    255.255.255.192 UG    0      0        0 eth0
```

```bash
$ sudo iptables -L
# ...
Chain cali-INPUT (1 references)
target     prot opt source               destination         
ACCEPT     ipv4 --  anywhere             anywhere             /* cali:PajejrV4aFdkZojI */ /* Allow IPIP packets from Calico hosts */ match-set cali40all-hosts-net src ADDRTYPE match dst-type LOCAL
DROP       ipv4 --  anywhere             anywhere             /* cali:_wjq-Yrma8Ly1Svo */ /* Drop IPIP packets from non-Calico hosts */
# ...
```

## Context

I want to enable IPIP traffic for non calico hosts in my worker node.
I am not sure why but my underline infrastructure requires IPIP traffic flow allow for passing load balancing health check in DSR mode.

## Your Environment

* calico/node version: v3.17.3
* Kubernetes: v1.17.12
* Operating System and version: CentOS 7 (kernel 3.10.0-1160.15.2.el7.x86_64)

**Comment by hongkunyoo** [comment]
@mgleung, Thank you for your help. It was very helpful. I disabled the IpInIpEnabled flag in Felix and confirmed that `iptables` **Drop non calico traffic** rule has been removed. If it wasn't for you I might had a hard problem finding out how to solve it. Unfortunately, there seems some other problems to connect to my LB related to my underline infrastructure. I will find out more on my own since my initial question has been solved.
Thank you.
